File: Finished/jumpGame.go
package leetcode

func canJump(nums []int) bool {
	dp := make([]bool, len(nums))
	dp[0] = true
	for i, val := range nums {
		if dp[i] == false {
			continue
		}

		for j := i; j <= i+val; j++ {
			if j >= len(nums) {
				return true
			}
			dp[j] = true
		}
	}

	return dp[len(nums)-1]
}

File: src/python/finished/jump_game.py
from typing import List

# https://leetcode.com/problems/jump-game/description


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        best = 0
        for i, n in enumerate(nums):
            if i > best:
                return False
            best = max(best, i + n)
        return True
